build/Band_Representations.jl

```diff
@@ -0,0 +1,247 @@
+using LinearAlgebra
+using Crystalline
+using StaticArrays
+
+function group_representative(symop::SymOperation, sg::SpaceGroup)
+    symop_xyzt=xyzt(symop)
+    for i in 1:size(sg)[1]
+        xyztiter=xyzt(sg[i])
+        if xyztiter==symop_xyzt
+            return i
+        end
+    end
+end
+
+function _find_isomorphic_parent_pointgroup(G)
+    D = dim(G)
+    ctᴳ = MultTable(G).table
+    @inbounds for iuclab in PGS_IUCs[D]
+        P = pointgroup(iuclab, D)
+        ctᴾ = MultTable(P).table
+        if ctᴳ == ctᴾ # bit sloppy; would miss ismorphisms that are "concealed" by row/column swaps
+            return P
+        end
+    end
+    return nothing # in case we didn't find any isomorphic parent
+end
+
+function in_site_symmetry(sitegroup::SiteGroup, symop::SymOperation)
+    #index= findfirst(x -> x==rotation(symop), rotation.(sitegroup)) 
+    index= findfirst(x -> x==xyzt(symop), xyzt.(sitegroup)) 
+    if index == nothing
+        return 0 
+    else 
+        return index
+    end
+end
+
+"This function gives the label for spacegroup(i, j)"
+function get_label(i::Int64, j::Int64)
+    return (label∘spacegroup)(i, j) 
+end
+
+"This function gives the site_symmetry irreps for the Site Symmetry Group"
+function site_symmetry_irreps(sitegroup::SiteGroup{D}) where D
+    site_parent_pointgroup=_find_isomorphic_parent_pointgroup(sitegroup)
+
+    get_pgirreps(site_parent_pointgroup.label, Val(D))
+end
+
+"The function below gives the character table for the site symmetry group provided"
+function site_symmetry_irreps_characters(sitegroup::SiteGroup{D}) where D
+    irreps=site_symmetry_irreps(sitegroup)
+    character_table=Matrix{Complex{Float64}}(undef, size(irreps[1].matrices)[1], size(irreps)[1] )
+    size(character_table)
+    for j in 1:size(irreps)[1]
+        for i in 1:size(irreps[1].matrices)[1]
+            character_table[i, j]=tr(irreps[j].matrices[i]) +0.00im +0.00
+        end
+    end
+character_table
+end
+
+
+"This function provides the induced band representation"
+function induced_band_representation(sitegroup::SiteGroup{D}, h::SymOperation{D}, ρ::Int64, k::KVec) where D
+    orbit_of_sitegroup=orbit(sitegroup, sitegroup.wp);
+    length_of_orbit=length(orbit_of_sitegroup)
+    #Now we sum over all the wyckoff positions in the orbit  
+    gα=cosets(sitegroup)
+    χ_G_k_G=0;
+    wp=sitegroup.wp;
+    wp_rvec=wp.qv.cnst
+    gα_inverse=gα[1]
+
+    transformed_k=transform(k, rotation(h)).cnst
+    for α′ in 1:length_of_orbit 
+
+        gα′=gα[α′]
+        tα′α′=rotation(h)*orbit_of_sitegroup[α′].qv.cnst+translation(h)-orbit_of_sitegroup[α′].qv.cnst
+        symop_t=SymOperation(one(Crystalline.SquareStaticMatrices.SqSMatrix{D,Float64}), -tα′α′ )
```

Comment:
```suggestion
        symop_t = SymOperation(-tα′α′)
```
(it does the same thing)